collect_app/src/test/java/org/odk/collect/android/configure/StructureAndTypeSettingsValidatorTest.java

```diff
@@ -0,0 +1,79 @@
+package org.odk.collect.android.configure;
+
+import org.json.JSONObject;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@SuppressWarnings("PMD.DoubleBraceInitialization")
+public class StructureAndTypeSettingsValidatorTest {
+
+    private final Map<String, Object> generalDefaults = new HashMap<String, Object>() {{
+        put("key1", true);
+    }};
+
+    private final Map<String, Object> adminDefaults = new HashMap<String, Object>() {{
+        put("key1", "default");
+    }};
+
+    private final StructureAndTypeSettingsValidator validator = new StructureAndTypeSettingsValidator(generalDefaults, adminDefaults);
+
+    @Test
+    public void forEmptySettings_returnsTrue() throws Exception {
+        assertThat(validator.isValid(emptySettings()), is(true));
+    }
+
+    @Test
+    public void forNonJSON_returnsFalse() {
+        assertThat(validator.isValid("blah"), is(false));
+    }
+
+    @Test
+    public void withoutGeneralObject_returnsFalse() throws Exception {
+        JSONObject json = emptySettingsObject();
+        json.remove("general"); // Remove mutates and returns removed item :(
+
+        assertThat(validator.isValid(json.toString()), is(false));
```

Comment:
I believe that Collect implicitly required both `general` and `admin` previously so good to keep consistent. I think we should follow up and make it only required to have one, not necessarily both to avoid having an empty set of settings. If that sounds reasonable I can file a low-priority issue (and corresponding doc update).

Reply:
Given they can just be empty it feels ok to require both? Requiring both does simplify logic on our side as well.

Reply:
Fair enough!